app/src/main/java/io/gnosis/safe/ui/settings/app/passcode/ChangePasscodeFragment.kt

```diff
@@ -0,0 +1,107 @@
+import io.gnosis.safe.ScreenId
+import io.gnosis.safe.databinding.FragmentPasscodeBinding
+import io.gnosis.safe.di.components.ViewComponent
+import io.gnosis.safe.ui.base.BaseStateViewModel
+import io.gnosis.safe.ui.base.SafeOverviewBaseFragment
+import io.gnosis.safe.ui.base.fragment.BaseViewBindingFragment
+import io.gnosis.safe.utils.showConfirmDialog
+import pm.gnosis.svalinn.common.utils.showKeyboardForView
+import pm.gnosis.svalinn.common.utils.visible
+import javax.inject.Inject
+
+class ChangePasscodeFragment : BaseViewBindingFragment<FragmentPasscodeBinding>() {
+
+    override fun screenId() = ScreenId.PASSCODE_CHANGE
+
+    @Inject
+    lateinit var viewModel: ChangePasscodeViewModel
+
+    override fun inflateBinding(inflater: LayoutInflater, container: ViewGroup?): FragmentPasscodeBinding =
+        FragmentPasscodeBinding.inflate(inflater, container, false)
+
+    override fun inject(component: ViewComponent) {
+        component.inject(this)
+    }
+
+    override fun onResume() {
+        super.onResume()
+        binding.input.showKeyboardForView()
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        viewModel.state.observe(viewLifecycleOwner, Observer {
+            when (val viewAction = it.viewAction) {
+                is ChangePasscodeViewModel.AllOwnersRemoved -> {
+
+                    findNavController().popBackStack(R.id.changePasscodeFragment, true)
+                    findNavController().currentBackStackEntry?.savedStateHandle?.set(SafeOverviewBaseFragment.PASSCODE_DISABLED_RESULT, true)
+                }
+                is BaseStateViewModel.ViewAction.ShowError -> {
+                    binding.errorMessage.setText(R.string.settings_passcode_owner_removal_failed)
+                    binding.errorMessage.visible(true)
+                }
+                is ChangePasscodeViewModel.PasscodeWrong -> {
+                    binding.errorMessage.setText(R.string.settings_passcode_wrong_passcode)
+                    binding.errorMessage.visible(true)
+                    binding.input.setText("")
+                }
+                is ChangePasscodeViewModel.PasscodeVerified -> {
+                    findNavController().navigate(
+                        ChangePasscodeFragmentDirections.actionPasscodeSettingsFragmentToChangeCreatePasscodeFragment(oldPasscode = binding.input.text.toString())
+                    )
+                }
+            }
+        })
+
+        with(binding) {
+            title.setText(R.string.settings_passcode_change_passcode)
+
+            status.visibility = View.INVISIBLE
+
+            createPasscode.setText(R.string.settings_passcode_enter_your_current_passcode)
+            backButton.setOnClickListener {
+                findNavController().popBackStack(R.id.changePasscodeFragment, true)
+            }
+
+            val digits = listOf(digit1, digit2, digit3, digit4, digit5, digit6)
```

Comment:
this stuff here regarding passcode input is duplicated in several fragments. It could be extracted to a base passcode fragment or some helper. I would add separate issue for this